include/base/variable.h

```diff
@@ -56,12 +56,12 @@ class Variable
    * all subdomains.
    */
   Variable (System * sys,
-            const std::string & var_name,
+            std::string var_name,
             const unsigned int var_number,
             const unsigned int first_scalar_num,
             const FEType & var_type) :
     _sys(sys),
-    _name(var_name),
+    _name(std::move(var_name)),
```

Comment:
There is probably an "Effective C++" guideline on this, but I'm curious exactly what the difference in performance is here? When passing by const reference we will make a copy (by calling the copy constructor) to store in the class. When passing by copy and moving, somehow a single copy is still made somewhere... the class doesn't steal the resources of any external string. So I see 1 string copy being made either way. Is the difference that we don't construct a *reference* in the new version? I have always thought of references as basically "free" or an "alias" but presumably there is some internal bookkeeping (similar to a pointer) that has to happen for them.

Reply:
If we construct `Variable(sys,str,m,n,fet)` then there's no difference in performance, we just make the same copy we would have made anyway, slightly earlier.  But now we can also choose to construct `Variable(sys,std::move(str),m,n,fet)`, and then the pass-by-value does a move construction on the value, and then the initialization here does another move, and nowhere in the whole sequence is there a copy made.

Kind of ridiculous in this case, I admit.  I'm mostly just trying to get into modern habits for when someday I'm writing code where either bigger strings or more string passing means being able to avoid some copies actually matters.

Reply:
OK, yeah makes sense for the case where the user constructs a string and then has no further use for it, so might as well let the `Variable` take over ownership and avoid any copies.